Sources/PointFree/Account/UpdateProfile.swift

```diff
@@ -0,0 +1,22 @@
+import Either
+import Foundation
+import HttpPipeline
+import Prelude
+import Tuple
+
+public struct ProfileData: Codable {
+  public let email: EmailAddress
+  public let name: String
+}
+
+let updateProfileMiddleware =
+  requireUser
+    <| { (conn: Conn<StatusLineOpen, Tuple2<Database.User, ProfileData?>>) -> IO<Conn<ResponseEnded, Data>> in
+      let (user, data) = lower(conn.data)
+
+      return pure(data)
+        .mapExcept(requireSome)
+        .flatMap { AppEnvironment.current.database.updateUser(user.id, $0.name, $0.email) }
```

Comment:
we should prob have them confirm their email change, but I think there's a very simple stateless way of doing it. we can send an email with a confirmation link that has a signed and encrypted value inside it, and we can use that to confirm the change.

no need to do it now, but I'll add a Trello card